web/src/components/molecules/core/SearchBar.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { styled } from '@material-ui/core/styles';
+import { ClearIcon, SearchButton, Container } from '../../atoms';
+import { InputField } from '../../molecules';
+import { OnChangeHandler, OnClickHandler } from '../../../types';
+import { flexStyle, flexRowStyle } from '../../../styles';
+
+export interface SearchBarProps {
+    input?: string;
+    onChange?: OnChangeHandler;
+    onClear?: OnClickHandler;
+    onSearch?: OnClickHandler;
+  };
+  
+export type SearchBar = React.FunctionComponent<SearchBarProps>;
+  
+export const SearchBar: SearchBar = ({ input, onChange, onClear, onSearch, ...other }) => {
+    const FlexContainer = React.useMemo(() => styled(({ ...props }) => <Container {...props}/>)(styles.flexContainer), []);
+    const FlexInputField = React.useMemo(() => styled(({ ...props }) => <InputField {...props}/>)(styles.flexInputField), []);
+    const FlexSearchButton = React.useMemo(() => styled(({ ...props }) => <SearchButton {...props}/>)(styles.flexSearchButton), []);
+
+    return (
+        <FlexContainer {...other}>
+            <FlexInputField input={input} icon={ClearIcon} onChange={onChange} onClick={onClear}/>
+            <FlexSearchButton onClick={onSearch}/>
+        </FlexContainer>
+    );
+};
+
+const styles = {
```

Comment:
the styles would be better in a shared file

Reply:
General design thinking here is to have shared styles in shared files (e.g. flex styles), but component specific styles in that component scope (in this case I think two components have the same 95/5 ratio of children, but IMO it's probably good to decouple them so we can style them separately if necessary/don't get style changes unexecptedly propagating around the place).

TLDR; I think this is OK as these styles are component specific so no need to import something that's only used in one file?

Reply:
styles in react are a rich source of discussion!
I just commented on this one because the same style set was used in another component and it looked like a case of re-use. Creating a flex-container component perhaps.
Fine as you are.